components/core/CMakeLists.txt

```
@@ -1,6 +1,14 @@
 cmake_minimum_required(VERSION 3.16.3)
+
+# Toolchain setup must come before the first project() call in the entire CMake buildsystem.
+# If CLP is not the top-level project, the following setup has no effect.
+include("cmake/Toolchains/utils.cmake")
```

```suggestion
include("${CMAKE_CURRENT_SOURCE_DIR}/cmake/Toolchains/utils.cmake")
```

Unless that's wrong for some reason?